quint/src/itf.ts

```diff
@@ -90,18 +119,34 @@ export function toItf(vars: string[], states: QuintEx[]): Either<string, ItfTrac
               return left('record: expected an even number of arguments, found:' + ex.args.length)
             }
             return merge(ex.args.map(exprToItf)).mapRight(kvs => {
-              let obj: any = {}
+              let obj: ItfRecord = {}
               chunk(kvs, 2).forEach(([k, v]) => {
-                obj[k] = v
+                if (typeof k === 'string') {
+                  obj[k] = v
+                } else {
+                  left(`Invalid record field: ${ex}`)
+                }
               })
               return obj
             })
           }
 
           case 'Map':
-            return merge(ex.args.map(exprToItf)).mapRight(pairs => {
-              return { '#map': pairs.map(p => p['#tup']) }
-            })
+            return merge(
+              // Convert all the entries of the map
+              ex.args.map(exprToItf)
+            ).chain(pairs =>
+              merge(
+                // Quint represents map entries as tuples, but in ITF they are 2 element arrays,
+                // so we unpack all the ITF tuples into arrays
```

Comment:
Should this be
```suggestion
                // so we unpack all the quint tuples into arrays
```
?

Reply:
Confusing, right :) But this actually unpacking ITF tuples. First all the arguments are sent to the toItf converter, which produces a list of itf tuples, but then we need to convert those int pairs of itf values, since itf doesn’t represent maps as a list of tuples, but ad a list of 2 element lists.

the logic could maybe be improved here, but I was trying not to get too far into fiddling with the previous code.

Reply:
Ah I see it now, thanks! I was not following the code on this part before, but from the previous comment line, this seemed wrong. Makes sense now!